packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1473,13 +1537,20 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
           canRequestFocus: _enabled,
           focusNode: focusNode,
           autofocus: widget.autofocus,
-          child: MouseRegion(
-            cursor: effectiveMouseCursor,
-            child: GestureDetector(
-              onTap: _enabled ? _handleTap : null,
-              behavior: HitTestBehavior.opaque,
-              child: result,
-            ),
+          child: InkWell(
+            onTap: _enabled ? _handleTap : null,
+            mouseCursor: effectiveMouseCursor,
+            enableFeedback: false,
+            excludeFromSemantics: true,
+            canRequestFocus: false,
+            child: widget._inputDecoration != null ?
+              InputDecorator(
+                decoration: widget._inputDecoration!,
+                isEmpty: widget._isEmpty!,
+                isFocused: widget._isFocused!,
+                child: result,
+              ) :
+              result,
           ),
```

Comment:
Nit.
Update `result` to `InputDecorator(..., child: result)` if `widget._inputDecoration != null` before returning. Hence, child can be set to `result` directly.

Reply:
Makes sense